storage/txcache/txCache.go

```diff
@@ -114,12 +123,85 @@ func (cache *TxCache) CountTx() int64 {
 	return cache.txByHash.counter.Get()
 }
 
+// Len is an alias for CountTx
+func (cache *TxCache) Len() int {
+	return int(cache.CountTx())
+}
+
 // CountSenders gets the number of senders in the cache
 func (cache *TxCache) CountSenders() int64 {
 	return cache.txListBySender.counter.Get()
 }
 
-// forEachSender iterates over the senders
+// forEachSender iterates over the senders in the cache
 func (cache *TxCache) forEachSender(function ForEachSender) {
 	cache.txListBySender.forEach(function)
 }
+
+// ForEachTransaction iterates over the transactions in the cache
+func (cache *TxCache) ForEachTransaction(function ForEachTransaction) {
+	cache.txByHash.forEach(function)
+}
+
+// Clear clears the cache
+func (cache *TxCache) Clear() {
+	cache.txListBySender.clear()
+	cache.txByHash.clear()
+}
+
+// Put is not implemented
+func (cache *TxCache) Put(key []byte, value interface{}) (evicted bool) {
+	panic("TxCache.Put is not implemented")
+}
+
+// Get gets a transaction by hash
+func (cache *TxCache) Get(key []byte) (value interface{}, ok bool) {
+	tx, ok := cache.GetByTxHash(key)
+	return tx, ok
+}
+
+// Has is not implemented
+func (cache *TxCache) Has(key []byte) bool {
+	panic("TxCache.Has is not implemented")
+}
+
+// Peek gets a transaction by hash
+func (cache *TxCache) Peek(key []byte) (value interface{}, ok bool) {
+	tx, ok := cache.GetByTxHash(key)
+	return tx, ok
+}
+
+// HasOrAdd is not implemented
+func (cache *TxCache) HasOrAdd(key []byte, value interface{}) (ok, evicted bool) {
+	panic("TxCache.HasOrAdd is not implemented")
+}
+
+// Remove is not implemented
+func (cache *TxCache) Remove(key []byte) {
+	panic("TxCache.Remove is not implemented")
+}
+
+// RemoveOldest is not implemented
+func (cache *TxCache) RemoveOldest() {
+	panic("TxCache.RemoveOldest is not implemented")
```

Comment:
maybe delete these panic stuff :). even if not implemented, these should be there, or delete the functions from the interface.